homeassistant/components/sun.py

```diff
@@ -59,10 +59,10 @@ def __init__(self, hass, location):
         """Initialize the sun."""
         self.hass = hass
         self.location = location
-        self._state = self.next_rising = self.next_setting = None
-        self.next_dawn = self.next_dusk = None
-        self.next_midnight = self.next_noon = None
-        self.solar_elevation = self.solar_azimuth = 0
+        self._state = self.next_rising = self.next_setting = STATE_UNKNOWN
```

Comment:
Just make `self.solar_elevation` and `self.solar_azimuth` `None` in the constructor. Leave the other properties alone. They will all be set before the entity is added to the state machine so the initial values don't need to be anything specific.